### PR TITLE
Update "Effective Dart" for optional new/const.

### PR DESCRIPTION
**Pull request by munificent**
- Remove unneeded "new"/"const" from excerpts (except for deliberately
  bad examples).
- Add guidelines for optional "new" and optional "const".
- Fix scripts/analyze-and-test-examples.sh now that the VM does not
  accept --preview-dart-2.

### CODE REVIEW

File: src/_guides/language/effective-dart/usage.md [hunk 1]
```diff
@@ -1079,12 +1078,7 @@ class Point {
...
 
-<!-- TODO(rnystrom): Uncomment this when all Dart tools support it. -->
-<!--
-
-**TODO: Use "###" for header. Removed so this doesn't go into TOC.**
-
-DON'T use `new`.
+### DON'T use `new`.
 
 Dart 2 makes the `new` keyword optional. Even in Dart 1, its meaning was never
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/usage.md]: Why no "bad" example?

Comment (kwalrath) [reply src/_guides/language/effective-dart/usage.md]: (I could add it for you, if you just want to be done with this.)

Comment (munificent) [reply src/_guides/language/effective-dart/usage.md]: Done.

**Comment by munificent** [comment]
Thanks!

**Pull request merged**